### ISSUE TITLE
GraphQLTransportWSHandler sends invalid errors message breaking `graphql-ws` clients on error (should be an array)

### ISSUE THREAD

**Issue by kevinvalk** [mirumee/ariadne#1053]
Lately I switched over to fully using the GraphQLTransportWSHandler for everything (including mutations and queries). However, I am receiving the following error on the client (`"graphql-ws": "^5.11.2",`).

```plain
index.ts:43 Error: "error" message expects the 'payload' property to be an array of GraphQL errors, but got {"message":".......","locations":[{"line":1,"column":36}],"extensions":{"exception":null}}
    at validateMessage (common.mjs:128:23)
    at parseMessage (common.mjs:168:12)
    at socket2.onmessage (client.mjs:202:37)
```

After checking `graphql-ws` source code and protocol specification  I think Apollo is not sending errors correctly:
https://github.com/enisdenjo/graphql-ws/blob/fbb763a662802a6a2584b0cbeb9cf1bde38158e0/src/utils.ts#L57-L66
https://github.com/enisdenjo/graphql-ws/blob/fbb763a662802a6a2584b0cbeb9cf1bde38158e0/PROTOCOL.md#error

The client expects an array of errors, but instead Apollo only sends a single error?
https://github.com/mirumee/ariadne/blob/3218cff61ce12dc649b1f425c26e23715c3241a1/ariadne/asgi/handlers/graphql_transport_ws.py#L318-L332

https://github.com/mirumee/ariadne/blob/3218cff61ce12dc649b1f425c26e23715c3241a1/ariadne/asgi/handlers/graphql_transport_ws.py#L368-L379

**Comment by rafalp** [comment]
I'll mark this as a bug and put this on my radar for next release.

**Comment by kevinvalk** [comment]
I only got to test this just now, but @rafalp the fix is not completely according spec and will still throw errors:
```
GraphQL websocket error:  Error: "error" message expects the 'payload' property to be an array of GraphQL errors, but got [{"errors":[{"message":"Unknown argument 'id' on field 'Mutation.test'.","locations":[{"line":2,"column":20}],"extensions":{"exception":null}}]}]
```
```json
{
    "type": "error",
    "id": "c243e373-4048-4108-a8d5-d5a23ed8e796",
    "payload": [
        {
            "errors": [
                {
                    "message": "Unknown argument 'id' on field 'Mutation.test'.",
                    "locations": [
                        {
                            "line": 2,
                            "column": 20
                        }
                    ],
                    "extensions": {
                        "exception": null
                    }
                }
            ]
        }
    ]
}
```


The [spec](https://github.com/enisdenjo/graphql-ws/blob/fbb763a662802a6a2584b0cbeb9cf1bde38158e0/PROTOCOL.md#error) mentions that when `type == 'error'` the payload is directly a list of errors, while currently it is a list of objects that contains the "errors" key with a list of errors.

``` typescript
import { GraphQLError } from 'graphql';

interface ErrorMessage {
  id: '<unique-operation-id>';
  type: 'error';
  payload: GraphQLError[];
}
```

**Comment by rafalp** [comment]
If I understand the spec right, if there's an error, instead of `NEXT` we should send `ERROR` message, because this protocol is not supporting partialy correct results?

**Comment by rafalp** [comment]
It seems to me that there are three places in `GraphQLTransportWSHandler` where errors can be returned to the client, and neither of those can create the error response you are mentioning:

https://github.com/mirumee/ariadne/blob/dfffa9018088566069b4f56fbee11ef875f70525/ariadne/asgi/handlers/graphql_transport_ws.py#L325

`type` is `ERROR` and `payload` is list of error objects, not `{"errors": [...]}`. Looks okay.

https://github.com/mirumee/ariadne/blob/dfffa9018088566069b4f56fbee11ef875f70525/ariadne/asgi/handlers/graphql_transport_ws.py#L374

`type` is `ERROR` and `payload` is a list of result responses. Looks like this is the culprit. But I'll need further study.

**Comment by rafalp** [comment]
I can reproduce this. This error only occurs for `query` and `mutation` operations.

**Comment by rafalp** [comment]
I've released 0.22 beta 1 with the fix.
